### PR TITLE
Simplify types and declare named exports

### PR DESCRIPTION
**Pull request by novascreen**
Fixes https://github.com/char0n/ramda-adjunct/issues/2306

This change removes the namespace wrapper from the type definitions along with the `exports =` export. 
Instead named exports are typed as such to reflect what's actually happening in the JS files.

TypeScript will allow us to re-export everything like this after the change:
```ts
export * from 'ramda-adjunct'
```

So far the only change i noticed is that VSCode will no longer auto-import `RA` when using Suggest, but since that was doing an import like this, that's probably not desired anyway:

```ts
import RA = require('ramda-adjunct');
```

However Quick Fix:
image
results in the expected import:
```ts
import * as RA from 'ramda-adjunct'
```

### CODE REVIEW

**Comment by char0n** [comment]
Looks good so far; let me know when you finished testing.

File: types/index.d.ts [hunk 1]
```diff
@@ -1,1553 +1,1775 @@
...
+/**
+ * Divides two numbers, where the second number is divided by the first number.
+ */
+export function divideNum(divisor: number, dividend: number): number;
+export function divideNum(divisor: number): (dividend: number) => number;
+
+/**
+ * Identity type.
+ */
+export type Identity = Function;
```
[the remark this answers is not in the document]
Comment (novascreen) [reply types/index.d.ts]: at least now i can re-export it, but this type was already not useful. I think it should be addressed separately

**Comment by novascreen** [comment]
in case you missed it @char0n it's ready for review 🙏

**Review by char0n — APPROVED** [review]
Thanks for contributing!

I did some testing using big project: https://github.com/swagger-api/apidom

Everything looks correct. I'm using [WebStorm](https://www.jetbrains.com/webstorm/) and everything works correctly and types pass tests. I'll move forward with the merge and release the week. We'll see if there will be some issues reported regarding the types.

So I guess we have now two configurations from two major IDEs. Moving forward with the merge and release during tommorow.

**Pull request merged**